Bunny.cpp

```diff
@@ -91,18 +91,22 @@ Bunny::~Bunny() {
 void Bunny::handleMessage(const Message<VECTOR2i> &message) {
     if (message.description == "teleported") {
         uint16_t currentFrame = World::instance()->currentFrame;
-        assert(currentFrame >= birth_ && currentFrame <= death_ && "Trying to teleport out of range");
+        assert(currentFrame >= birth_
+            && currentFrame <= death_
+            && "Trying to teleport out of range");
         newState_ = BUNNY_STATE_PLAYING;
 
-        Bunny *newBunny = new Bunny(BUNNY_STATE_RECORDING, currentFrame, 0xFFFF);
-        BunnyPhysicsComponent *physics = newBunny->getComponent<BunnyPhysicsComponent *>();
+        Bunny *newBunny;
+        newBunny = new Bunny(BUNNY_STATE_RECORDING, currentFrame, 0xFFFF);
+        BunnyPhysicsComponent *physics;
+        physics = newBunny->getComponent<BunnyPhysicsComponent *>();
 
         if (physics) {
             addHandler(static_cast<MessageHandler<VECTOR2i> *>(physics));
             dispatchMessage(message);
```

Comment:
Hmm. For some reason, adding a handler, dispatching a message, and removing a handler seems strange to me. We should discuss.